### ISSUE TITLE
[92] Revisiting Multi-Scale Feature Fusion for Semantic Segmentation

### ISSUE THREAD

**Issue by dhkim0225** [dhkim0225/1day_1paper#121]
[paper](https://arxiv.org/abs/2203.12683)

정말 단순하다.
segmentation 에는 P9까지 쓰면 좋다구~~~
굳이 ASPP 같은 모듈 쓸 필요 없다. (이게 cost 가 크다)
끝.

![image](https://user-images.githubusercontent.com/16400591/164726195-74bad6f9-6c6c-4452-9efa-138d64004871.png)

성능 비교.
![image](https://user-images.githubusercontent.com/16400591/164726059-3e2847e1-70b1-47a2-8aa5-3aea581e0e92.png)

확실히 semantic segmentation 같은 task 는 receptive field 가 중요한 역할을 하는 것으로 보인다.
![image](https://user-images.githubusercontent.com/16400591/164726303-e2623479-02ff-45af-8a82-07aa00ba8f73.png)

decoder 는 변형된 bifpn 형태를 가져갔다.
최종적으로 weighted sum 을 하는 형태.
![image](https://user-images.githubusercontent.com/16400591/164726700-185d8bd2-920b-420c-ba88-0f8bb746aa6e.png)
단순 weighted sum 은 아니고 learnable weight w 들을 두고, softmax 수행
![image](https://user-images.githubusercontent.com/16400591/164733308-b73d6e88-8125-4449-8358-c6c6b7081f3c.png)

channel wise learnable attention 느낌. (learnable query)
test 시에는 정해진 weight를 기준으로 softmax 를 제거할 수 있음 (fuse layer)

# 추가 속도 향상
backbone 은 efficientnet을 썼는데 느렸다고 한다.
MBConv 대신 Effv2 에서 layer 앞쪽에서만 썼던 fused-MBConv 를 모든 레이어에 활용했다.
SiLU(Swish-1) 는 ReLU 로 바꿨고
속도 면에서 가성비가 구린 SE module 은 모두 없앴다. 

# Cityscape
![image](https://user-images.githubusercontent.com/16400591/164741074-875c5fef-6883-450d-9583-fa5ea9b100b6.png)
![image](https://user-images.githubusercontent.com/16400591/164741214-b8cb61e1-035d-47a5-b317-7a8a4b31da50.png)
![image](https://user-images.githubusercontent.com/16400591/164741374-8ece1ee7-673b-4a5c-875d-806727d5a374.png)

pretraining == Mapillary Vistas set 으로 500 epochs 학습 
self-training == coarse annotated data 에 coarse label 안쓰고 confidence 0,5 이상인 애들만 갖고 label 만들어서 training 수행
coarse 영역 label 을 이용해서 negative 정보도 줄 수 있음.
![image](https://user-images.githubusercontent.com/16400591/164741446-e07364e1-5536-4736-a9f3-3b3aa2fbe0d6.png)

# ADE-20K
![image](https://user-images.githubusercontent.com/16400591/164741152-7802ab4e-cc99-479c-a7a4-49f2d8dd54c2.png)
